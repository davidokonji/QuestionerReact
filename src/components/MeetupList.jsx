import React, { Fragment } from 'react';
import moment from 'moment';
import { confirmAlert } from 'react-confirm-alert';
import Button from './Button';
import { calendar, location } from '../Assets';

const MeetupList = ({
  title, location: meetupLocation, happeningOn, id, deleteOne
}) => {
  const date = moment(happeningOn).format('dddd, MMMM Do YYYY');
  const deleteHandler = () => {
    confirmAlert({
      title: 'Delete meetup',
      message: `Are you sure to delete ${title}`,
      buttons: [
        {
          label: 'Yes',
          onClick: () => deleteOne(id)
        },
        {
          label: 'No',
          onClick: () => {}
        }
      ],
      closeOnEscape: true,
      closeOnClickOutside: true,
    });
  };
  return (
    <Fragment>
      <div className='col-8 col-md-10 border border-left-0 border-right-0'>
        <div className='row'>
          <p className='col-12 ml-3'>
            { title }
          </p>
        </div>
        <div className='row'>
          <p className='col-3 col-md-1'>
            <img src={calendar} alt='date' className='h-75 icon' />
          </p>
          <p className='col-9 col-md-9'>
            { date }
          </p>
        </div>
        <div className='row'>
          <p className='col-3 col-md-1'>
            <img src={location} alt='location' className='h-75 icon' />
          </p>
          <p className='col-9 col-md-9'>
            { meetupLocation }
          </p>
        </div>
      </div>
      <div className='col-4 col-md-2 mt-3 p-0'>
        <Button
          text='Delete'
          styles='px-3 py-1 my-2 btn-danger col-10 rounded-top rounder-bottom'
          event={deleteHandler}
        />
        <Button
          text='View'
          styles='px-3 py-1 my-2 btn-info col-10 rounded-top rounder-bottom'
          event={() => window.location.assign(`/meetup/${id}`)}
        />
      </div>
    </Fragment>
  );
};

export default MeetupList;
