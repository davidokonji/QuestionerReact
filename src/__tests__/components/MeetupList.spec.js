import React from 'react';
import { shallow } from 'enzyme';
import MeetupList from '../../components/MeetupList';

const data = {
  title: '',
  location: '',
  happeningOn: '',
  id: ''
};

describe('Meetup list component', () => {
  it('should render component correctly', () => {
    const wrapper = shallow(<MeetupList {...data} />);

    expect(wrapper).toMatchSnapshot();
  });
});
