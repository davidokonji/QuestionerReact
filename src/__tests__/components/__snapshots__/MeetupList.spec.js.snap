// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Meetup list component should render component correctly 1`] = `
<Fragment>
  <div
    className="col-10 border border-left-0 border-right-0"
  >
    <div
      className="row"
    >
      <p
        className="col-12 ml-3"
      />
    </div>
    <div
      className="row"
    >
      <p
        className="col-1"
      >
        <img
          alt="date"
          className="h-75 icon"
          src={
            Object {
              "canInstrument": true,
              "createTransformer": [Function],
              "getCacheKey": [Function],
              "process": [Function],
            }
          }
        />
      </p>
      <p
        className="col-9"
      >
        Invalid date
      </p>
    </div>
    <div
      className="row"
    >
      <p
        className="col-1"
      >
        <img
          alt="location"
          className="h-75 icon"
          src={
            Object {
              "canInstrument": true,
              "createTransformer": [Function],
              "getCacheKey": [Function],
              "process": [Function],
            }
          }
        />
      </p>
      <p
        className="col-9"
      />
    </div>
  </div>
  <div
    className="col-2 mt-3"
  >
    <Button
      event={[Function]}
      styles="px-3 py-1 my-2 btn-danger col-10 rounded-top rounder-bottom"
      text="Delete"
    />
    <Button
      event={[Function]}
      styles="px-3 py-1 my-2 btn-info col-10 rounded-top rounder-bottom"
      text="View"
    />
  </div>
</Fragment>
`;
